code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -62,6 +62,34 @@
 		for(var/mob/O in viewers(user, 3))
 			O.show_message(text("<span class='notice'>\The [src] has been broken by [] with an electromagnetic card!</span>", user), 1, text("You hear a faint electrical spark."), 2)
 			return
+
+/obj/item/weapon/storage/lockbox/emp_act(severity)
+	for(var/obj/O in src)
+		if(istype(O, /obj/item/weapon/gun/energy/stunrevolver))
+			explosion(O.loc,0,1,4) //fuck your shitty stun revolvers
+			qdel(src)
+		var/obj/item/weapon/stock_parts/cell/C
+		for(C in O)
+			C.corrupt()
+			if(prob(80)) //cell is probably kill
+				C.corrupt()
+				C.corrupt()

```

Comment:
the doubling is intentional. at least, if i understand corrupt right. the first corrupt is guaranteed to half the cell's max charge, then its 80 prob to 1/8th the cell, completely ruining it and leaving a dud gun.